docs/src/main/asciidoc/datasource.adoc

```diff
@@ -436,69 +421,77 @@ AgroalDataSource usersDataSource;
 
-Some databases like H2 and Derby are commonly used in "embedded mode" as a facility to run quick integration tests.
+You will usually specify the `db-kind` property or explicitly enable Dev Services by setting `quarkus.datasource."name".devservices.enabled=true`.
 
-Our suggestion is to use the real database you intend to use in production; container technologies made this simple enough so you no longer have an excuse. Still, there are sometimes
-good reasons to also want the ability to run quick integration tests using the JVM powered databases,
-so this is possible as well.
+[[in-memory-databases]]
+=== Testing with in-memory databases
+
+Some databases like H2 and Derby are commonly used in the _embedded mode_ as a facility to run integration tests quickly.
 
-It is important to remember that when configuring Derby to use the embedded engine,
-this will work as usual in JVM mode but such an application will not compile into a native executable, as the Quarkus extensions only cover for making the JDBC client code compatible with the native compilation step: embedding the whole database engine into a native executable is currently not implemented.
+It is recommended to use the real database you intend to use in production, but it’s possible to use JVM-powered databases for the scenarios when the ability to run quick integration tests is required.
 
-Compiling H2 into a native-image is now possible (it didn't use to be) but this is considered experimental; even if this is technically possible,
-we would still recommend to consider carefully why you would want to embed H2 within your native-image.
+Note that while configuring Derby to use the embedded engine works as usual in JVM mode, such an application will not compile into a native executable, because the Quarkus extensions only cover making the JDBC client code compatible with the native compilation step.
+Embedding the entire database engine into a native executable is currently not implemented.
 
-If you plan to run such integration tests in the JVM exclusively, it will of course work as usual.
+Compiling H2 into a native image is now possible, but this is considered experimental; even if technically possible,
+We recommend carefully considering why you want to embed H2 within your native image.
 
-If you want the ability to run such integration test in both JVM and/or native executables, we have some cool helpers for you: just add either `@QuarkusTestResource(H2DatabaseTestResource.class)` or `@QuarkusTestResource(DerbyDatabaseTestResource.class)` on any class in your integration tests, this will make sure the test suite starts (and stops) the embedded database into a separate process as necessary to run your tests.
+Running integration tests with embedded Derby or H2 in the JVM mode works as expected in this scenario, but they will not work in native mode.
```

Comment:
Didn't we just state that it would work with H2?

Reply:
Well, I think we had 3 instances like this in the doc and I was really not sure what the original author thought with this. I would always stay what works when and vice versa.

Reply:
Okay, but this line is new and FWIU it directly contradicts the line literally just above... ?

Above:

> Compiling H2 into a native image is [...] technically possible

Here:

> Running integration tests with embedded Derby or H2 [...] will not work in native mode


---

To be constructive, from what I understand:

* With Derby, it doesn't work.
* With H2, it used not to work, but nowadays it does.

Reply:
Thank you! Applying.